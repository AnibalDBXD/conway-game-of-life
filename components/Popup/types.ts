export interface IPopup {
    trigger: JSX.Element;
    children: React.ReactNode;
    title: string;
}