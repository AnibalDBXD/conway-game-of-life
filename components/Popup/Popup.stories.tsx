/* eslint-disable @typescript-eslint/no-empty-function */
import React from "react";
import { Story, Meta } from "@storybook/react";

import Popup from ".";
import { IPopup } from "./types";
import Button from "../Button";

export default {
  title: "Components/Popup",
  component: Popup,
  parameters: {
    docs: {
      description: {
        component: "Play or pause button"
      }
    },
  }
} as Meta;

const Template: Story<IPopup> = (args: IPopup) => <Popup {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  title: "Popup title",
  children: "lorem impsum lorem impsumlorem impsum lorem impsum lorem impsum lorem impsum lorem impsum lorem impsum lorem impsum lorem impsum",
  trigger: <Button>Trigger</Button>
};