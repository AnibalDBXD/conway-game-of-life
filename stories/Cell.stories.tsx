/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { Story, Meta } from '@storybook/react';

import Cell from '../components/Table/Cell';
import { ICellProps } from '../components/Table/Cell/types';

export default {
  title: 'Pages/Home/Table/Cell',
  component: Cell,
  parameters: {
    docs: {
      description: {
        component: "Table's cell",
      },
    },
  },
} as Meta;

const Template: Story<ICellProps> = (args: ICellProps) => <Cell {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  isActive: false,
};

export const Secondary = Template.bind({});
Primary.args = {
  isActive: true,
};
