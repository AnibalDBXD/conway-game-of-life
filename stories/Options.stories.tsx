/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { Story, Meta } from '@storybook/react';
import GameProvider from '../context/Game/GameProvider';

import Options from '../components/Options';

export default {
  title: 'Pages/Home/Options',
  component: Options,
  parameters: {
    docs: {
      description: {
        component: 'Table options',
      },
    },
  },
} as Meta;

const Template: Story = () => (
  <GameProvider>
    <Options />
  </GameProvider>
);

export const Primary = Template.bind({});
