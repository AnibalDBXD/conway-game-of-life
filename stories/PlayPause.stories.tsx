/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { Story, Meta } from '@storybook/react';

import PlayPause from '../components/PlayPause';
import { IPlayPause } from '../components/PlayPause/types';

export default {
  title: 'Components/PlayPause',
  component: PlayPause,
  parameters: {
    docs: {
      description: {
        component: 'Play or pause button',
      },
    },
  },
} as Meta;

const Template: Story<IPlayPause> = (args: IPlayPause) => (
  <PlayPause {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  pause: false,
};
