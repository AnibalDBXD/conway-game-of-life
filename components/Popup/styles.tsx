import Popup from 'reactjs-popup';
import styled from 'styled-components';
import { COLORS } from '../../common/styles';

export const StyledPopup = styled(Popup)`
  &-overlay {
    background-color: rgba(0, 0, 0, 0.5);
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;

export const StyledContainer = styled.section`
  overflow: auto;
  background-color: ${COLORS.WHITE};
  border-radius: 16px;
  width: 50vw;
  max-height: 90vh;
  padding: 16px;
  box-shadow: 2px 13px 23px -3px rgba(0, 0, 0, 0.79);
  animation: 0.5s ease-out slideInFromTop forwards;

  @keyframes slideInFromTop {
    0% {
      transform: translateY(-100%);
    }

    100% {
      transform: translateY(0);
    }
  }
`;

export const StyledTitleContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
`;

export const StyledTitle = styled.h1`
  margin: 0;
  font-weight: bold;
`;

export const StyledClose = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  position: static;
  right: -10px;
  top: -10px;
  position: absolute;
  background: 0;
  border: 0;
  width: 2rem;
  height: 2rem;
  cursor: pointer;
`;
