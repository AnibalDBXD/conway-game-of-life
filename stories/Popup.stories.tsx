/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { Story, Meta } from '@storybook/react';

import Popup from '../components/Popup';
import { IPopup } from '../components/Popup/types';
import Button from '../components/Button';

export default {
  title: 'Components/Popup',
  component: Popup,
  parameters: {
    docs: {
      description: {
        component: 'Play or pause button',
      },
    },
  },
} as Meta;

const Template: Story<IPopup> = (args: IPopup) => <Popup {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  title: 'Popup title',
  children:
    'lorem impsum lorem impsumlorem impsum lorem impsum lorem impsum lorem impsum lorem impsum lorem impsum lorem impsum lorem impsum',
  trigger: <Button>Trigger</Button>,
};
