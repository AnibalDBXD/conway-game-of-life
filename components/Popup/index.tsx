import { StyledContainer, StyledPopup, StyledClose, StyledTitleContainer, StyledTitle } from "./styles";
import { IPopup } from "./types";
import Close from '../../common/icons/close.svg';
import { Children, cloneElement, isValidElement } from "react";

const Popup = ({ trigger, children, title }: IPopup): JSX.Element => {
  return (
    <StyledPopup
      closeOnDocumentClick
      closeOnEscape
      modal
      trigger={trigger}
    >
      {(close: () => void): JSX.Element => (
        <StyledContainer>
          <StyledTitleContainer>
            <StyledTitle>{title}</StyledTitle>
            <StyledClose onClick={close}><Close /></StyledClose>
          </StyledTitleContainer>
          {Children.map(children, child => {
            if (isValidElement(child)) {
              return cloneElement(child, {close});
            }
            return child;
          })}
        </StyledContainer>
      )}
    </StyledPopup>
  );
};

export default Popup;
