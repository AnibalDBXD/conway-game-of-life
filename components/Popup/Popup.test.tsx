import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Primary } from './Popup.stories';
import "jest-styled-components";

describe('<Popup />', () => {
  test("Render", () => {
    const { asFragment } = render(<Primary {...Primary.args} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("Render on popup", () => {
    const { asFragment, getByText } = render(<Primary {...Primary.args} />);
    const button = getByText("Trigger");

    fireEvent.click(button);
    expect(asFragment()).toMatchSnapshot();
  });
  test("Should be show title and childrens and click button close", () => {
    const { getByText, queryByText, getAllByRole } = render(<Primary {...Primary.args} />);
    const button = getByText("Trigger");
    fireEvent.click(button);

    getByText("Popup title");
    getByText("lorem impsum lorem impsumlorem impsum lorem impsum lorem impsum lorem impsum lorem impsum lorem impsum lorem impsum lorem impsum");

    fireEvent.click(getAllByRole("button")[0]);
    expect(queryByText("Popup title")).toBeNull();
  });
});