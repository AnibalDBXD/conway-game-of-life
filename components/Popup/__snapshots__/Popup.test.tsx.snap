// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Render 1`] = `
<DocumentFragment>
  .c0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  background-color: transparent;
  border: 2px solid #FFFFFF;
  border-radius: 8px;
  color: #FFFFFF;
  font-weight: bold;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.1s ease;
  -webkit-transition: transform 0.1s ease;
  transition: transform 0.1s ease;
}

.c0:hover {
  border-color: #FFFFFFAE;
  color: #FFFFFFAE;
}

.c0:active {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

<button
    aria-describedby="popup-1"
    class="c0"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`<Input /> Render on popup 1`] = `
<DocumentFragment>
  .c0 {
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  background-color: transparent;
  border: 2px solid #FFFFFF;
  border-radius: 8px;
  color: #FFFFFF;
  font-weight: bold;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.1s ease;
  -webkit-transition: transform 0.1s ease;
  transition: transform 0.1s ease;
}

.c0:hover {
  border-color: #FFFFFFAE;
  color: #FFFFFFAE;
}

.c0:active {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

<button
    aria-describedby="popup-2"
    class="c0"
  >
    Trigger
  </button>
</DocumentFragment>
`;
